From: David Bauer <mail@david-bauer.net>
Date: Thu, 17 Nov 2022 22:53:33 +0100
Subject: ath79: combine UniFi AC dual firmware-partitions

In order to maximize the available space on UniFi AC boards using a
dual-image partition layout, combine the two OS partitions into a single
partition.

This allows users to access more usable space for additional packages.

Don't limit the usable image size to the size of a single OS partition.
The initial installation has to be done with an older version of OpenWrt
in case the generated image exceeds the space of a single kernel
partition in the future.

Signed-off-by: David Bauer <mail@david-bauer.net>
(cherry picked from commit e4a76673ff4f655ba0698d9edb9efbacd0f82fff)

diff --git a/target/linux/ath79/dts/qca9563_ubnt_unifiac.dtsi b/target/linux/ath79/dts/qca9563_ubnt_unifiac.dtsi
index 2e407c348642cf5145f79c15288856fa141b0512..6704ec983b0beed5421a59f663cf8db6d650f559 100644
--- a/target/linux/ath79/dts/qca9563_ubnt_unifiac.dtsi
+++ b/target/linux/ath79/dts/qca9563_ubnt_unifiac.dtsi
@@ -69,17 +69,12 @@
 			};
 
 			partition@70000 {
+				/* Combine kernel0 & kernel1 */
 				label = "firmware";
-				reg = <0x070000 0x790000>;
+				reg = <0x070000 0xf20000>;
 				compatible = "denx,uimage";
 			};
 
-			partition@800000 {
-				label = "kernel1";
-				reg = <0x800000 0x790000>;
-				read-only;
-			};
-
 			partition@f90000 {
 				label = "bs";
 				reg = <0xf90000 0x020000>;
diff --git a/target/linux/ath79/image/generic-ubnt.mk b/target/linux/ath79/image/generic-ubnt.mk
index 7642c59e02848741996d6f9dcaf923d5dbaf6880..d6898c79d6b4d51d19813781e086db8086537318 100644
--- a/target/linux/ath79/image/generic-ubnt.mk
+++ b/target/linux/ath79/image/generic-ubnt.mk
@@ -221,7 +221,7 @@ TARGET_DEVICES += ubnt_unifi
 define Device/ubnt_unifiac
   DEVICE_VENDOR := Ubiquiti
   SOC := qca9563
-  IMAGE_SIZE := 7744k
+  IMAGE_SIZE := 15488k
   DEVICE_PACKAGES := kmod-ath10k-ct ath10k-firmware-qca988x-ct
 endef
 
